packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -75,12 +75,17 @@ exports.sourceNodes = async (
   // Remove deleted entries & assets.
   // TODO figure out if entries referencing now deleted entries/assets
   // are "updated" so will get the now deleted reference removed.
+
+  function deleteContentfulNode (node) {
+    const id = normalize.fixId(node.sys.id)
```

Comment:
isn't `normalize.fixId` already done here - https://github.com/axe312ger/gatsby/blob/3058f95dc020524dec1242da3523a1e9a5f73e5d/packages/gatsby-source-contentful/src/fetch.js#L75-L86 (`fixIds` seem to traverse passed object and run `fixId` on `id` fields) ?

Reply:
Also the id is passed to your new function not the node. You need to fix the id and then use that to get the node.

Reply:
node is passed there - lines that maps node -> id are deleted ( lines 79, 82 ), but I checked again and we can skip normalizing id here (it's already normalized)